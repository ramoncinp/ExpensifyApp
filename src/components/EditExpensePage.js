import React from "react";
import { connect } from "react-redux";
import ExpenseForm from "./ExpenseForm";
import { editExpense, removeExpense } from "../actions/expenses";

export class EditExpensePage extends React.Component {
  onEditExpense = expense => {
    this.props.editExpense(expense.id, expense);
    this.props.history.push('/');
  }

  onRemoveExpense = () => {
    this.props.removeExpense({ id: this.props.expense.id });
    this.props.history.push('/');
  }

  render () {
    return (
      <div>
      <ExpenseForm
        expense={this.props.expense}
        onSubmit={this.onEditExpense}
      />
      <button
        onClick={this.onRemoveExpense}
      >
        Remove
      </button>
    </div>
    );
  };
};

const mapsStateToProps = (state, props) => {
  return {
    expense: state.expenses.find(
      expense => expense.id === props.match.params.id
    )
  };
};

const mapDispatchToProps = (dispatch, props) => ({
  editExpense: (id, expense) => dispatch(editExpense(id, expense)),
  removeExpense: (data) => dispatch(removeExpense(data))
});

export default connect(mapsStateToProps, mapDispatchToProps)(EditExpensePage);
