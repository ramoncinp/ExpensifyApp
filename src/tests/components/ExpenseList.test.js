import React from "react";
import { ExpenseList } from "../../components/ExpenseList";
import { shallow } from "enzyme";
import expenses from "../fixtures/expenses";

test("should render ExpenseList with expenses", () => {
  const wrapper = shallow(<ExpenseList expenses={expenses} />);
  expect(wrapper).toMatchSnapshot();
});

test("sholud render ExpenseLit with empty message", () => {
    const wrapper = shallow(<ExpenseList expenses={[]}/>);
    expect(wrapper).toMatchSnapshot();
});