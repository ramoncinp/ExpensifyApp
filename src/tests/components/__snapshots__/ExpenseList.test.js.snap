// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sholud render ExpenseLit with empty message 1`] = `
<div>
  <p>
    No expenses
  </p>
</div>
`;

exports[`should render ExpenseList with expenses 1`] = `
<div>
  <ExpenseListItem
    amount={500}
    createdAt={0}
    description="Gum"
    id="1"
    key="1"
    note=""
  />
  <ExpenseListItem
    amount={1200}
    createdAt={-345600000}
    description="Bread"
    id="2"
    key="2"
    note=""
  />
  <ExpenseListItem
    amount={1500}
    createdAt={345600000}
    description="Cookies"
    id="3"
    key="3"
    note=""
  />
</div>
`;
